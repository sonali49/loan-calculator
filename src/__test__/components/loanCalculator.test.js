import { render, screen } from "@testing-library/react";
import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { unmountComponentAtNode } from "react-dom";
import LoanCalculator from "../../components/loanCalculator";
import LoanCalculatorTemplate from "../../components/loanCalculatorTemplate";

Enzyme.configure({ adapter: new Adapter() });

describe("Loan Calculator", () => {
  let container = null;
  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("renders without crashing", () => {
    render(<LoanCalculator />);
  });

  it("Should render loan calculator template", () => {
    const handleChange = jest.fn();
    const component = render(
      <LoanCalculatorTemplate
        monthlyIncome={500000}
        requestedAmount={20000000}
        loanTerm={38}
        selectedChildren="NONE"
        coApplicant="NONE"
        handleChange={handleChange()}
        disabled={false}
      />
    );
    expect(component.container).toMatchSnapshot();
  });
  it("Submit button should be disabled if fields are empty", () => {
    const handleChange = jest.fn();
    render(
      <LoanCalculatorTemplate
        monthlyIncome={500000}
        requestedAmount={20000000}
        loanTerm={38}
        selectedChildren="NONE"
        coApplicant="NONE"
        handleChange={handleChange()}
        disabled={true}
      />
    );

    const submitButton = screen.getByRole("button");
    expect(submitButton).toBeDisabled();
  });
});
