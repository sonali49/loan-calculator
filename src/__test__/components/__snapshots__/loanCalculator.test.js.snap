// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loan Calculator Should render loan calculator template 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      for="monthlyIncome"
      required=""
    >
      Monthly Income
    </label>
    <input
      class=""
      name="monthlyIncome"
      type="text"
      value="500000"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      for="requestedAmount"
      required=""
    >
      Requested Amount
    </label>
    <input
      class=""
      name="requestedAmount"
      type="text"
      value="20000000"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      for="loanTerm"
      required=""
    >
      Loan Term(in months)
    </label>
    <input
      class=""
      name="loanTerm"
      type="text"
      value="38"
    />
  </div>
  <div
    class="form-group"
  >
    <label
      for="selectedChildren"
    >
      Children
    </label>
    <select
      class=""
      name="selectedChildren"
    >
      <option
        value=""
      >
        --Choose an option--
      </option>
      <option
        value="NONE"
      >
        NONE
      </option>
      <option
        value="SINGLE"
      >
        SINGLE
      </option>
      <option
        value="MULTIPLE"
      >
        MULTIPLE
      </option>
    </select>
  </div>
  <div
    class="form-group"
  >
    <label
      for="coApplicant"
    >
      Co-applicant
    </label>
    <select
      class=""
      name="coApplicant"
    >
      <option
        value=""
      >
        --Choose an option--
      </option>
      <option
        value="NONE"
      >
        NONE
      </option>
      <option
        value="SINGLE_BORROWER"
      >
        SINGLE_BORROWER
      </option>
      <option
        value="MULTIPLE_BORROWERS"
      >
        MULTIPLE_BORROWERS
      </option>
    </select>
  </div>
  
  <button
    type="submit"
  >
    Submit
  </button>
</div>
`;
